### ISSUE TITLE
Stack creation issue in CNF - "Error occurred while GetObject. S3 Error Code: AuthorizationHeaderMalformed. S3 Error Message: The authorization header is malformed; the region 'us-east-1' is wrong; expecting 'us-west-2'"

### ISSUE THREAD

**Issue by DataDirk** [theemadnes/aws_iot_button_lap_counter#1]
Hi Alex, 
great work! I came across your project, and since I'm just starting out with AWS and had bought a button, I thought I give it a shot and reproduce your project. 

I've upload your CNF 'template.yaml' into the stack creator, it asked me for a name and the IoT button DNS, so far so good.

However, I'm then running into a rollback issue, though, when creating the stack. Please see events below.

  | 20:35:02 UTC+0200 | ROLLBACK_IN_PROGRESS | AWS::CloudFormation::Stack | IoTbutton-dashboard | The following resource(s) failed to create: [DashboardApiFunction, ButtonPressFunction]. . Rollback requested by user.
-- | -- | -- | -- | --

  | 20:35:00 UTC+0200 | CREATE_FAILED | AWS::Lambda::Function | ButtonPressFunction | Error occurred while GetObject. S3 Error Code: AuthorizationHeaderMalformed. S3 Error Message: The authorization header is malformed; the region 'us-east-1' is wrong; expecting 'us-west-2'

  | 20:35:00 UTC+0200 | CREATE_FAILED | AWS::Lambda::Function | DashboardApiFunction | Error occurred while GetObject. S3 Error Code: AuthorizationHeaderMalformed. S3 Error Message: The authorization header is malformed; the region 'us-east-1' is wrong; expecting 'us-west-2'

I'm working in region 'us-east-1'. I can't figure out what the issue is. 
Would you have any advise what I could try?

Many thanks in advance,
Dirk

**Comment by DataDirk** [comment]
just created the stack via logging into us-west-2 without any issues. Consequently, the API is also only present in us-west-2, and not in us-east-1.
Problem is that I had created a thing for my iot-button in us-east-1, and it looks like I cannot reuse from another region, but need to create a new thing for it.

I just don't understand why I cannot do this in us-east-1. Any idea what the root cause is, and if anything can be done to create this application in us-east-1?

**Comment by theemadnes** [comment]
yep - ill dig in to it. generally this can be solved by creating a map to point to the correct resource location based on the region the stack is being created in. I'll make the tweaks in a day or so, and provide some notes on what i did to fix it.
